### PR TITLE
Integrate nginxreceiver

### PR DESCRIPTION
**Pull request by nslaughter**
**Description:** 
The nginx receiver was not previously included in the registration of components in `internal/components/components.go` and therefore was not building in the collector.

**Link to tracking Issue:** 
[issue 108 in the releases repo](https://github.com/open-telemetry/opentelemetry-collector-releases/issues/108) explains. 

**Testing:** 
There were pre-existing unit and integration tests of the package which were passing and continue to pass after changes. I also added an entry for this receiver in `internal/components/receivers_test.go,` which tests the receiver lifecycle. I also used the functionality to send metrics in a live environment.

**Documentation:** 
I have not expanded upon existing documentation.

### CODE REVIEW

**Comment by pmm-sumo** [comment]
Thank you for the contribuion @nslaughter, could you sign the CLA per the [contributing guidelines](https://github.com/open-telemetry/community/blob/main/CONTRIBUTING.md#sign-the-cla)?

**Comment by nslaughter** [comment]
> Thank you for the contribuion @nslaughter, could you sign the CLA per the [contributing guidelines](https://github.com/open-telemetry/community/blob/main/CONTRIBUTING.md#sign-the-cla)?

Indeed. It's in progress with company approver.

**Review by codeboten — CHANGES_REQUESTED** [review]
Looks like something happen during rebase, please fix

**Comment by nslaughter** [comment]
Thank you. It's fixed.

**Comment by dmitryax** [comment]
Please resolve conflicts

**Comment by nslaughter** [comment]
I'm tending to CI issues blocking now. ty.

**Comment by dmitryax** [comment]
@nslaughter please also sign CLA to be able to contribute in OTel

File: go.mod [hunk 1]
```diff
@@ -104,6 +104,7 @@ require (
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/memcachedreceiver v0.50.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver v0.50.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mysqlreceiver v0.50.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/nginxreceiver v0.0.0-00010101000000-000000000000
```
[the remark this answers is not in the document]
Comment (nslaughter) [reply go.mod:107]: fixed

**Comment by nslaughter** [comment]
Are there issues with any load tests or do failures indicate a problem with some change?
I'm wondering as the PR doesn't change Carbon or SignalFx related code, but the TestMetric10kDPS tests are failing for each.

File: internal/components/receivers_test.go [hunk 2]
```diff
@@ -162,6 +162,10 @@ func TestDefaultReceivers(t *testing.T) {
 		{
 			receiver: "mysql",
 		},
+		{
+			receiver: "nginx",
+		},
+
```
Comment (codeboten) [inline internal/components/receivers_test.go:168]: nit:
```suggestion
```

**Review by codeboten — APPROVED** [review]
Just a minor comment, otherwise this looks good

File: go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (nslaughter) [reply go.mod]: Unfortunately that is my mistake and it is now in [this new commit](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/9572/commits/e7f047503a7bd3bd7850cd302ce35e20c1641bfb).

**Review by dmitryax — APPROVED** [review]

**Pull request merged**